game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua

```diff
@@ -0,0 +1,90 @@
+modifier_aura_item_upgrade = class(ModifierBaseClass)
+
+if IsServer() then
+  function modifier_aura_item_upgrade:OnCreated( kv )
+    self.ItemName = kv.ItemName
+    self.PlayerId = kv.PlayerId
+
+    self.AuraItems =
+    {
+      "item_greater_guardian_greaves_",
+      "item_greater_travel_boots_",
+      "item_ancient_jangoo_of_endurance_",
+      "item_assault_",
+      "item_crimson_pipe_",
+      "item_helm_of_the_dominator_",
+      "item_lucience_",
+      "item_mekansm_",
+      "item_pipe_",
+      "item_radiance_",
+      "item_ring_of_aquila_",
+      "item_shivas_guard_",
+      "item_urn_of_sorcery_",
+      "item_vladmir_",
+    }
+
+    local hero = PlayerResource:GetPlayer(self.PlayerId):GetAssignedHero()
+
+    print(self.ItemName)
+    -- Only remove the item if it has aura upgrade
+    for _, value in ipairs(self.AuraItems) do
+      if string.find(self.ItemName, value) then
+        -- only remove the item if it is in a active slot
+        for i = DOTA_ITEM_SLOT_1, DOTA_ITEM_SLOT_6 do
+          local item = hero:GetItemInSlot(i)
+          if item then
+            if item:GetName() == self.ItemName then
+              self.ItemSlot = i
+              hero:RemoveItem(item)
```

Comment:
Should probably save the handle to the item, use `TakeItem` then use `AddItem` to give it back again so that you don't destroy the item and create a new one. Example of this can be found in `courier.lua`.

Reply:
Valve documentation sucks....
I saw the TakeItem but on API it just says 

TakeItem | void TakeItem(handle hItem) | Hidden item